examples fail to build with boost 1.83+

The boost API changed with boost 1.80+, see also [here](https://www.boost.org/doc/libs/1_80_0/libs/timer/doc/index.html)

```
FAILED: examples/CMakeFiles/chainiksolverpos_lma_demo.dir/chainiksolverpos_lma_demo.cpp.o 
/usr/bin/x86_64-pc-linux-gnu-g++  -I/usr/include/eigen3 -I/var/tmp/portage/sci-libs/orocos_kdl-1.5.1/work/orocos_kinematics_dynamics-1.5.1/orocos_kdl/src -I/var/tmp/portage/sci-libs/orocos_kdl-1.5.1/work/orocos_kinematics_dynamics-1.5.1/orocos_kdl/models -I/var/tmp/portage/sci-libs/orocos_kdl-1.5.1/work/orocos_kdl-1.5.1_build/src  -O2 -pipe -march=znver3 -ggdb -MD -MT examples/CMakeFiles/chainiksolverpos_lma_demo.dir/chainiksolverpos_lma_demo.cpp.o -MF examples/CMakeFiles/chainiksolverpos_lma_demo.dir/chainiksolverpos_lma_demo.cpp.o.d -o examples/CMakeFiles/chainiksolverpos_lma_demo.dir/chainiksolverpos_lma_demo.cpp.o -c /var/tmp/portage/sci-libs/orocos_kdl-1.5.1/work/orocos_kinematics_dynamics-1.5.1/orocos_kdl/examples/chainiksolverpos_lma_demo.cpp
In file included from /var/tmp/portage/sci-libs/orocos_kdl-1.5.1/work/orocos_kinematics_dynamics-1.5.1/orocos_kdl/examples/chainiksolverpos_lma_demo.cpp:59:
/usr/include/boost/timer.hpp:21:3: error: #error This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
   21 | # error This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
      |   ^~~~~
In file included from /usr/include/boost/config/header_deprecated.hpp:18,
                 from /usr/include/boost/timer.hpp:24:
/usr/include/boost/timer.hpp:25:1: note: ‘#pragma message: This header is deprecated. Use the facilities in <boost/timer/timer.hpp> instead.’
   25 | BOOST_HEADER_DEPRECATED( "the facilities in <boost/timer/timer.hpp>" )
      | ^~~~~~~~~~~~~~~~~~~~~~~

```

--- a/orocos_kdl/examples/CMakeLists.txt
+++ b/orocos_kdl/examples/CMakeLists.txt
@@ -8,7 +8,12 @@ IF(ENABLE_EXAMPLES)
   TARGET_LINK_LIBRARIES(trajectory_example orocos-kdl)
   
   add_executable(chainiksolverpos_lma_demo chainiksolverpos_lma_demo.cpp )
-  TARGET_LINK_LIBRARIES(chainiksolverpos_lma_demo orocos-kdl orocos-kdl-models)
+  find_package(Boost REQUIRED)
+  IF(${Boost_VERSION_MACRO} LESS 108300)
+    TARGET_LINK_LIBRARIES(chainiksolverpos_lma_demo orocos-kdl orocos-kdl-models)
+  ELSE()
+    TARGET_LINK_LIBRARIES(chainiksolverpos_lma_demo boost_timer orocos-kdl orocos-kdl-models)
+  ENDIF()
 
 ENDIF(ENABLE_EXAMPLES)  
 

--- a/orocos_kdl/examples/chainiksolverpos_lma_demo.cpp
+++ b/orocos_kdl/examples/chainiksolverpos_lma_demo.cpp
@@ -56,7 +56,12 @@ estimate of shortest time per invposkin (ms) 0.155544
 #include <chainfksolverpos_recursive.hpp>
 #include <utilities/utility.h>
 
+#include <boost/version.hpp>
+#if BOOST_VERSION < 108300
 #include <boost/timer.hpp>
+#else
+#include <boost/timer/timer.hpp>
+#endif
 
 /**
  * tests the inverse kinematics on the given kinematic chain for a
@@ -64,7 +69,11 @@ estimate of shortest time per invposkin (ms) 0.155544
  * \TODO provide other examples.
  */
 void test_inverseposkin(KDL::Chain& chain) {
+#if BOOST_VERSION < 108300
     boost::timer timer;
+#else
+    boost::timer::cpu_timer timer;
+#endif
     int num_of_trials = 1000000;
     int total_number_of_iter = 0;
     int n = chain.getNrOfJoints();
@@ -159,7 +168,12 @@ void test_inverseposkin(KDL::Chain& chain) {
     std::cout << "max. trans. difference after solving " << max_trans_diff << std::endl;
     std::cout << "min. rot. difference after solving " << min_rot_diff << std::endl;
     std::cout << "max. rot. difference after solving " << max_rot_diff << std::endl;
+#if BOOST_VERSION < 108300
     double el = timer.elapsed();
+#else
+    boost::timer::cpu_times const ct(timer.elapsed());
+    double el = ct.user / 1e9;
+#endif
     std::cout << "elapsed time " << el << std::endl;
     std::cout << "estimate of average time per invposkin (ms)" << el/num_of_trials*1000 << std::endl;
     std::cout << "estimate of longest time per invposkin (ms) " << el/total_number_of_iter*max_num_of_iter *1000 << std::endl;

